src/auto/attr_iterator.rs

```diff
@@ -46,8 +46,8 @@ impl AttrIterator {
 
     pub fn range(&mut self) -> (i32, i32) {
         unsafe {
-            let mut start = mem::uninitialized();
-            let mut end = mem::uninitialized();
+            let mut start = mem::MaybeUninit::uninit().assume_init();
+            let mut end = mem::MaybeUninit::uninit().assume_init();
```

Comment:
In is not good example of "Incorrect usage of this method:" ? 
See https://doc.rust-lang.org/std/mem/union.MaybeUninit.html#method.assume_init

Reply:
Yes and no. We want to have something in an uninitialised state as far as I understand it and we want it to be assumed as initialised so we can use it. But maybe I misunderstood the problem?

Reply:
The choice we have is to call `assume_init` *after* the C function was called.

Reply:
Seems correct usage is call after C function.
Also on `rustc 1.34.0` "MaybeUninit" is still "unstable feature" 😢

Reply:
Yes the way how it is here now is undefined behaviour and defeats the purpose of `MaybeUninit`. You have to assert/assume that it's initialized after calling the C function.

You'd do
```rust
let mut foo = MaybeUninit::uninit();
ffi::foo_bar(foo.as_mut_ptr());
let foo = foo.assume_init();
```

That's defined behaviour and all good then.

Reply:
>  Also on rustc 1.34.0 "MaybeUninit" is still "unstable feature" cry

It's stable in 1.36. As this is for the next gtk-rs release, updating the minimum required version to 1.36 seems fine to me.

Reply:
> ...
> That's defined behaviour and all good then.

So we all agree. I'll update the gir PR later on.

> It's stable in 1.36. As this is for the next gtk-rs release, updating the minimum required version to 1.36 seems fine to me.

Seems fine to me as well.